TDL-09-shell-EC2/deploy_instance.sh

```diff
@@ -0,0 +1,229 @@
+print_color "green" "-----------------------------Check input-------------------------------"
+
+# if less than 2 parameters from command line, print usage() and exit
+if [[ $# -lt 2 || $# -gt 3 ]];  then
+  usage 'all'
+  exit
+fi
+
+# check whehter ostype is a valid option, if ostype is an invalid ostype, print usage() and exit 
+list_include_item "${LIST_OSTYPE[*]}" "$1"
+result=$?
+if (( result == 1 )); then
+  usage 'ostype'
+  exit
+fi
+ostype=$1
+
+# check whether 1 <= agedtime <=7, otherwise iprint usage() and exit
+if (( $2 <1 || $2 > 7)); then
+  usage 'agedtime'
+  exit
+fi
+agedtime=$2
+
+# check whether instance_type is valid, otherwise iprint usage() and exit
+if [[ $# -eq 3 ]];  then
+  list_include_item "${LIST_INSTANCE_TYPE[*]}" "$3"
+  result=$?
+  if (( result == 1 )); then
+    usage 'instance_type'
+    exit
+  fi
+  instance_type=$3
+fi
+echo "input correct"
+
+# Find the latest image id for ostype and print out the image_id, image_name
+print_color "green" "------------------Find the latest image id for ostype -----------------"
+case "$ostype" in
+  amzn)
+    image_id=$(aws ssm get-parameters --names /aws/service/ami-amazon-linux-latest/amzn-ami-hvm-x86_64-gp2 \
+    --query 'Parameters[0].[Value]' --output text)
+    # image_id=$(aws ec2 describe-images  --owners 'amazon'  --filters 'Name=description,Values=Amazon Linux AMI*' \
+    # --query 'sort_by(Images, &CreationDate)[-1].[ImageId]'  --output 'text')
+    ;;
+  amzn2)
+    image_id=$(aws ssm get-parameters --names /aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2 \
+      --query 'Parameters[0].[Value]' --output text)
+    ;;
+  centos6)
+    image_id=$(aws ec2 describe-images --owners 679593333241 \
+      --filters \
+        Name=name,Values='CentOS Linux 6 x86_64 HVM EBS*' \
+        Name=architecture,Values=x86_64 \
+        Name=root-device-type,Values=ebs \
+      --query 'sort_by(Images, &Name)[-1].ImageId' \
+      --output text)
+    ;;
+  centos7)
+    image_id=$(aws ec2 describe-images --owners aws-marketplace \
+      --filters \
+        Name=name,Values='CentOS Linux 7 x86_64 HVM*' \
+        Name=product-code,Values=aw0evgkw8e5c1q413zgy5pjce \
+        Name=architecture,Values=x86_64 \
+        Name=root-device-type,Values=ebs \
+      --query 'sort_by(Images, &CreationDate)[-1].[ImageId]' \
+      --output text)
+    ;;
+  ubuntu)
+    image_id=$(aws ec2 describe-images \
+      --owners 099720109477 \
+      --filters Name=name,Values=ubuntu/images/hvm-ssd/ubuntu-xenial-16.04-amd64-server-* \
+      --query 'sort_by(Images,&CreationDate)[-1].ImageId' --output text)
+    ;;
+esac
+echo "Find the latest image of $ostype: ${image_id}"
+ 
+#  Create EC2 instance for osype with instance_type with below settings:
+print_color "green" "-------------------------Create EC2 instance---------------------------"
+# set variables
+KEY_NAME='da-key'
+VPC_NAME='da-c02-vpc'
+VPC_ID=$(aws ec2 describe-vpcs \
```

Comment:
It is better to define these variables at the very beginning of the script so you can update the code later easily in the future.

Reply:
Ok.